spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -335,6 +335,7 @@ public DeltaWriter<InternalRow> createWriter(int partitionId, long taskId) {
       OutputFileFactory dataFileFactory =
           OutputFileFactory.builderFor(table, partitionId, taskId)
               .format(context.dataFileFormat())
+              .operationId(context.queryId())
```

Comment:
@rdblue just FYI since you had this in your initial version. I have not added this to the `deleteFileFactory` below because otherwise delete writers are overwriting the data files (due to having the exact same file name)

Reply:
https://github.com/apache/iceberg/blob/8c6adf6e5e17603025d23b2012aa576c071ff269/core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java#L90 shows how the file name is being determined, and in the cases where the data file was overwritten, `partitionId / taskId / operationId` were all the same (since we manually set the same `operationId` as for data files - previously the `operationId` for data+delete files was randomly generated).

Maybe we could add a different suffix into the name generation to indicate that it's a data/delete file (although I'm not sure if there are any implications to this)

Reply:
@nastra, why wasn't the `fileCount.incrementAndGet()` taking care of the problem? Were the two paths using different output file factories or the same one?

Reply:
Okay, I see below that they were using a different factory.

Reply:
yes correct, they were using different output file factories